### ISSUE TITLE
opening Vinyl causes screen to turn off/turn on in loop until i reboot phone

### ISSUE THREAD

**Issue by ellisonch** [VinylMusicPlayer/VinylMusicPlayer#853]
**Describe the bug**
After updating to 1.5.6 (on F-Droid), opening Vinyl causes my phone's screen to turn off and on, in a loop, every few seconds.  The phone becomes unusable until I restart the phone.  After restarting, opening Vinyl again causes the on/off loop to happen again.  Opening Vinyl is the only thing that causes this loop.

I uninstalled, then reinstalled version 1.5.5.  I was able to open Vinyl afterwards and everything worked right at first, but after I closed Vinyl and reopened it, it now does the screen looping thing.

**To reproduce**
1. Open Vinyl
2. Wait a couple of seconds
3. Screen will turn off/on repeatedly, every 2 seconds or so

**Device info**
Please complete the following information. Or if you get directed here from the app, you can paste the collected 
device info.

 - Device: Moto G Power 2022
 - Android Version: 12
 - Vinyl Music Player Version: 1.5.6

**Comment by isaiaismaele** [comment]
I believe the issue may be related to the last system update, since I started experiencing problems after that, and since I share the same device model and Android version of @ellisonch.

Device: Moto G Power 2022
Android Version: 12
**Updated to version: STAS32.79-77-28-50-7**

Details say that this update contains Android security patches up to October 5, 2023.

Downgrading to either versions 1.5.4 or 1.5.5 of Vinyl did not solve the problem. After each new installation the application started without issues, built the media library, and played songs. But the second time I opened it, the System UI crashed and entered the same loop.

**Comment by ellisonch** [comment]
> Downgrading to either versions 1.5.4 or 1.5.5 of Vinyl did not solve the problem. After each new installation the application started without issues, built the media library, and played songs. But the second time I opened it, the System UI crashed and entered the same loop.

Yes, this describes my experience as well.

**Comment by ellisonch** [comment]
@soncaokim I can do whatever if you tell me what to do.  I am not familiar with android development, but I am a programmer.

Whatever happens needs to be capturable without any further input on the device, because after I open vinyl, I have to restart the phone to do literally anything.

**Comment by soncaokim** [comment]
I'm exploring here, so please bear with me.

First proposal: 
- After restart, where the phone is still usable, can you please go to Settings and set enable Collect crash report (bottom of the page).
- if there are errors inside Vinyl, it would capture the error and pops up a notification. Please reproduce the crash loop.
- post restart, please tell me if you have any notification telling Vinyl is crashed.

**Comment by ellisonch** [comment]
In order to get to vinyl's settings, I had to uninstall/reinstall vinyl (because once it starts crashing, any time you open the app it crashes the phone).

I tried to be thorough in describing everything I did in order:

1. uninstalled 1.6.1
2. reinstalled 1.6.1
3. open 1.6.1
4. it asks for, and I give it "photo and media" privilege
5. it starts rebuilding my library
6. I went to settings and enabled crash reports
7. closed vinyl
8. reopened vinyl
9. at this point, it doesn't crash, which surprises me (it usually crashes on start).  so i click on the hamburger menu in the top left and it crashes the phone
10. rebooted the phone
11. upon reboot, there were no notifications
12. opened vinyl and it immediately crashes the phone (no need, or even any chance to click anything.  it crashes the phone _immediately_)
13. rebooted the phone
14. no notifications

**Comment by soncaokim** [comment]
After step 3, it should ask you the permission for notification as well, right?

I'm afraid this has something beyond Vinyl. Do you have the issue with other music app, those with offline local library?

Second proposal - beware this relies on a Android system feature and might capture information beyond Vinyl. I dont know if your OS will capture any personal info. **If you are not comfortable to capture that info or share it to a stranger on the internet, please dont go further**

- Enable Developer settings on you phone - at the OS lebel, outside Vinyl. You can search for it for detailed steps, but basically it consists of repeatedly tapping on About / Version inside Settings until a short message tells you that Developer options are enabled.
- go back to Android settings, you should see Developer options (depend on the vendor/OS version, it might not be immediately shown - on my phone it's under System).
- inside Developer options, look if you have Bug report. If yes, tapping on it will collect systems info, including past crashes, and build a zip file. Once done, you'll have the chance to send it to yourself, open it, select only part that you think is relevant and related to the crash.

I cannot give you precise instruction on each step since it may vary depending on OS/vendor. Please document yourself first (google or android resources). And again, if you are not comfortable then dont take the risk.

**Comment by ellisonch** [comment]
> After step 3, it should ask you the permission for notification as well, right?

It does not.

> zip file

So I did what you said, but I couldn't make heads or tails of how that bug report file was organized.  So instead, I figured out how to use adb and connected to my phone and grabbed logs using `adb logcat -v time`.

When the crash happens, I get about a billion of these in the logfile (at least hundreds a second):

```
12-08 17:13:44.405 W/System.err(26225): java.lang.NumberFormatException: empty String
12-08 17:13:44.405 W/System.err(26225): 	at jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:1842)
12-08 17:13:44.405 W/System.err(26225): 	at jdk.internal.math.FloatingDecimal.parseFloat(FloatingDecimal.java:122)
12-08 17:13:44.405 W/System.err(26225): 	at java.lang.Float.parseFloat(Float.java:485)
12-08 17:13:44.405 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.tagging.ReplayGainTagExtractor.parseFloat(Unknown Source:8)
12-08 17:13:44.405 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.tagging.ReplayGainTagExtractor.parseId3Tags(Unknown Source:122)
12-08 17:13:44.405 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.tagging.ReplayGainTagExtractor.setReplayGainValues(Unknown Source:22)
12-08 17:13:44.405 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.tagging.TagExtractor.extractTags(Unknown Source:203)
12-08 17:13:44.405 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.Discography.addSong(Unknown Source:23)
12-08 17:13:44.405 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.Discography.getOrAddSong(Unknown Source:40)
12-08 17:13:44.405 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.Discography.syncWithMediaStore(Unknown Source:78)
12-08 17:13:44.405 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.SyncWithMediaStoreAsyncTask.doInBackground(Unknown Source:17)
12-08 17:13:44.405 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.SyncWithMediaStoreAsyncTask.doInBackground(Unknown Source:2)
12-08 17:13:44.405 W/System.err(26225): 	at android.os.AsyncTask$3.call(AsyncTask.java:394)
12-08 17:13:44.405 W/System.err(26225): 	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
12-08 17:13:44.405 W/System.err(26225): 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:305)
12-08 17:13:44.405 W/System.err(26225): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
12-08 17:13:44.405 W/System.err(26225): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
12-08 17:13:44.405 W/System.err(26225): 	at java.lang.Thread.run(Thread.java:1012)
```


There are also some that look like this:

```
12-08 17:13:43.977 W/System.err(26225): java.lang.NumberFormatException: For input string: "12345-1234-123-123-123456"
12-08 17:13:43.977 W/System.err(26225): 	at jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:2054)
12-08 17:13:43.977 W/System.err(26225): 	at jdk.internal.math.FloatingDecimal.parseFloat(FloatingDecimal.java:122)
12-08 17:13:43.977 W/System.err(26225): 	at java.lang.Float.parseFloat(Float.java:485)
12-08 17:13:43.977 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.tagging.ReplayGainTagExtractor.parseFloat(Unknown Source:8)
12-08 17:13:43.977 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.tagging.ReplayGainTagExtractor.parseId3Tags(Unknown Source:122)
12-08 17:13:43.977 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.tagging.ReplayGainTagExtractor.setReplayGainValues(Unknown Source:22)
12-08 17:13:43.977 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.tagging.TagExtractor.extractTags(Unknown Source:203)
12-08 17:13:43.977 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.Discography.addSong(Unknown Source:23)
12-08 17:13:43.977 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.Discography.getOrAddSong(Unknown Source:40)
12-08 17:13:43.977 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.Discography.syncWithMediaStore(Unknown Source:78)
12-08 17:13:43.977 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.SyncWithMediaStoreAsyncTask.doInBackground(Unknown Source:17)
12-08 17:13:43.977 W/System.err(26225): 	at com.poupa.vinylmusicplayer.discog.SyncWithMediaStoreAsyncTask.doInBackground(Unknown Source:2)
12-08 17:13:43.977 W/System.err(26225): 	at android.os.AsyncTask$3.call(AsyncTask.java:394)
12-08 17:13:43.977 W/System.err(26225): 	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
12-08 17:13:43.977 W/System.err(26225): 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:305)
12-08 17:13:43.977 W/System.err(26225): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
12-08 17:13:43.977 W/System.err(26225): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
12-08 17:13:43.977 W/System.err(26225): 	at java.lang.Thread.run(Thread.java:1012)
```

(I changed the number it's complaining about to 12345-1234-123-123-123456 since I don't know what it means, but there are others in different formats, like 12345678-123-1234-123-12345 and probably others)

There may be other problems, but I see so many of those I can't really sift through to find anything more particular.

**Comment by ellisonch** [comment]
So upon further investigation, it looks like these happen even when the phone doesn't crash.  It seems the way you scan mp3s involves throwing those exceptions.  Are you throwing exceptions as part of the normal operation in parsing mp3s?  If so, you could probably speed up rescanning by avoiding the throw/try/catch way of parsing bad strings.

I'll see if I can figure out a way to avoid these getting in the way of reporting the problem and get back to you.  Maybe i'll let it finish rescan my library before i restart the program the first time.  The problem is, I have a huge library.

**Comment by ellisonch** [comment]
@soncaokim

Ok, without really understanding any of this, it looks like the ultimate problem is related to Vinyl's notifications:

```
12-08 18:11:40.020 D/InterruptionStateProvider(25439): No alerting: filtered notification: 0|com.poupa.vinylmusicplayer|1|null|10543
12-08 18:11:40.022 D/InterruptionStateProvider(25439): No alerting: filtered notification: 0|com.poupa.vinylmusicplayer|1|null|10543
12-08 18:11:49.509 E/AndroidRuntime(25640): FATAL EXCEPTION: SysUiBg
12-08 18:11:49.509 E/AndroidRuntime(25640): Process: com.android.systemui, PID: 25640
12-08 18:11:49.509 E/AndroidRuntime(25640): java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.media.MediaMetadata.getString(java.lang.String)' on a null object reference
12-08 18:11:49.509 E/AndroidRuntime(25640): 	at com.android.systemui.media.MediaDataManager.loadMediaDataInBg(MediaDataManager.kt:580)
12-08 18:11:49.509 E/AndroidRuntime(25640): 	at com.android.systemui.media.MediaDataManager.access$loadMediaDataInBg(MediaDataManager.kt:103)
12-08 18:11:49.509 E/AndroidRuntime(25640): 	at com.android.systemui.media.MediaDataManager$loadMediaData$1.run(MediaDataManager.kt:343)
12-08 18:11:49.509 E/AndroidRuntime(25640): 	at android.os.Handler.handleCallback(Handler.java:978)
12-08 18:11:49.509 E/AndroidRuntime(25640): 	at android.os.Handler.dispatchMessage(Handler.java:104)
12-08 18:11:49.509 E/AndroidRuntime(25640): 	at android.os.Looper.loopOnce(Looper.java:238)
12-08 18:11:49.509 E/AndroidRuntime(25640): 	at android.os.Looper.loop(Looper.java:357)
12-08 18:11:49.509 E/AndroidRuntime(25640): 	at android.os.HandlerThread.run(HandlerThread.java:85)
```

It seems that somehow certain kinds of notifications can actually crash android's entire UI system (see, e.g., https://stackoverflow.com/q/64186578/2877032).  I think that's what's happening.  It fits with what I see, as well; as I mentioned in my first comment, it's not that the phone hangs.  Instead, the screen goes black, then it turns back on, then it goes black again.  So it makes sense that it's android's UI that's crashing.  It takes a few seconds to show up again, then it goes black, then it shows up again... and that's what I see in the logs.  This FATAL EXCEPTION happens every few seconds in the log.

I've pasted above all that's in the log about what vinyl is doing.  If you need to add more logging to help figure out what's going on, I can try doing this again after you've instrumented the code.

**Comment by soncaokim** [comment]
Hi, thanks for the effort. Will have a look this weekend.

**Comment by soncaokim** [comment]
> It seems the way you scan mp3s involves throwing those exceptions. Are you throwing exceptions as part of the normal operation in parsing mp3s

These `NumberFormatException` shows that you have ReplayGain tags in your library that the app cannot understand. ReplayGain will not work, you can see that the app has not managed to extract those values in the song Detail dialog. 

We can ignore this part - we print the errors in the log but that's benign.

**Comment by soncaokim** [comment]
One comment mentions that the issue starts with the system upgrade. What was the OS version before please - where the app functions properly? Was that Android 12 too?

**Comment by vojtech2** [comment]
Hi - I'm having exactly the same problem. I have LineageOS 19.1 (Android 12L on Pixel 3a) and for me it used to work with a previous weekly nightly build of the OS (still Android 12) but some weeks ago it stopped. I'm pretty sure an OS update broke it, not a Vinyl update.

Currently I'm on 19-20231010-NIGHTLY-sargo which I believe is the final one for Lineage OS 19.1 for this device. Sadly I'm not sure which exact build broke it.

**Comment by ravinrabbid** [comment]
I've observed the same behavior on a Xperia XZ1 Comapact with LineageOS 19.1 (19-20231015) after upgrading to a release including the October security patches.

Today I upgraded to the November security patches (LineageOS 19-20231122)  and could **not** reproduce the crash since then.

**Comment by ellisonch** [comment]
> One comment mentions that the issue starts with the system upgrade. What was the OS version before please - where the app functions properly? Was that Android 12 too?

I know it worked on November 8th and that it was broken on the 14th.  Looking at my phone's "system update" tool, it says that I updated to "S3RQS32.20-42-10-12-5-4" on November 9th.  According to the history, my previous version was "S3RQS32.20-42-10-12-5-3" which I started using September 6th, and I believe is also Android 12.  This matches what I see on https://www.verizon.com/support/motorola-moto-g-power-2022-update/

**Comment by soncaokim** [comment]
This will be hard to circumvent, and even if we manage to do something in Vinyl I'm afraid that will not be maintainable...

Nevertheless, I've made a hack here https://github.com/VinylMusicPlayer/VinylMusicPlayer/commit/3713e4a0847f164a5400e0c2cba6bce09040d20a - to skip showing the cover image to Android.

This is based on the 1.6.2 version

Can you please test if this build has any effect on the crashing situation? thanks
https://github.com/VinylMusicPlayer/VinylMusicPlayer/suites/18915390874/artifacts/1104747535

**Comment by ellisonch** [comment]
I uninstalled, and then installed 1.6.2.  I had to switch from f-droid to the play store, since 1.6.2 isn't on f-droid yet.

I get the same behavior.  The first time I run it, it rebuilds my library and plays music just fine as long as I want.  After closing Vinyl and restarting it, it crashes the phone.  After rebooting the phone, any time I open the app, it crashes the phone.

```
12-10 14:43:07.218 D/InterruptionStateProvider( 7848): No alerting: filtered notification: 0|com.poupa.vinylmusicplayer|1|null|10546
12-10 14:43:07.221 D/InterruptionStateProvider( 7848): No alerting: filtered notification: 0|com.poupa.vinylmusicplayer|1|null|10546
12-10 14:43:07.222 E/AndroidRuntime( 7848): FATAL EXCEPTION: SysUiBg
12-10 14:43:07.222 E/AndroidRuntime( 7848): Process: com.android.systemui, PID: 7848
12-10 14:43:07.222 E/AndroidRuntime( 7848): java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.media.MediaMetadata.getString(java.lang.String)' on a null object reference
12-10 14:43:07.222 E/AndroidRuntime( 7848): 	at com.android.systemui.media.MediaDataManager.loadMediaDataInBg(MediaDataManager.kt:580)
12-10 14:43:07.222 E/AndroidRuntime( 7848): 	at com.android.systemui.media.MediaDataManager.access$loadMediaDataInBg(MediaDataManager.kt:103)
12-10 14:43:07.222 E/AndroidRuntime( 7848): 	at com.android.systemui.media.MediaDataManager$loadMediaData$1.run(MediaDataManager.kt:343)
12-10 14:43:07.222 E/AndroidRuntime( 7848): 	at android.os.Handler.handleCallback(Handler.java:978)
12-10 14:43:07.222 E/AndroidRuntime( 7848): 	at android.os.Handler.dispatchMessage(Handler.java:104)
12-10 14:43:07.222 E/AndroidRuntime( 7848): 	at android.os.Looper.loopOnce(Looper.java:238)
12-10 14:43:07.222 E/AndroidRuntime( 7848): 	at android.os.Looper.loop(Looper.java:357)
12-10 14:43:07.222 E/AndroidRuntime( 7848): 	at android.os.HandlerThread.run(HandlerThread.java:85)
```

I'd also like to point out that the first time the app runs, it's showing info panels and notifications just fine.  It's only after restarting vinyl that vinyl crashes the phone.

**Comment by soncaokim** [comment]
Any chance to test the custom build I posted in the previous comment?

**Comment by ellisonch** [comment]
> Any chance to test the custom build I posted in the previous comment?

Ah, I misunderstood.  I thought you wanted me to update to 1.6.2.

I've installed your apk manually.  I even checked the about menu to make sure I was using your special version.  Same behavior.

```
12-10 15:56:40.937 D/InterruptionStateProvider( 1846): No alerting: filtered notification: 0|com.poupa.vinylmusicplayer.ci|1|null|10548
12-10 15:56:40.938 D/InterruptionStateProvider( 1846): No alerting: filtered notification: 0|com.poupa.vinylmusicplayer.ci|1|null|10548
--------- beginning of crash
12-10 15:56:40.943 E/AndroidRuntime( 1846): FATAL EXCEPTION: SysUiBg
12-10 15:56:40.943 E/AndroidRuntime( 1846): Process: com.android.systemui, PID: 1846
12-10 15:56:40.943 E/AndroidRuntime( 1846): java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.media.MediaMetadata.getString(java.lang.String)' on a null object reference
12-10 15:56:40.943 E/AndroidRuntime( 1846): 	at com.android.systemui.media.MediaDataManager.loadMediaDataInBg(MediaDataManager.kt:580)
12-10 15:56:40.943 E/AndroidRuntime( 1846): 	at com.android.systemui.media.MediaDataManager.access$loadMediaDataInBg(MediaDataManager.kt:103)
12-10 15:56:40.943 E/AndroidRuntime( 1846): 	at com.android.systemui.media.MediaDataManager$loadMediaData$1.run(MediaDataManager.kt:343)
12-10 15:56:40.943 E/AndroidRuntime( 1846): 	at android.os.Handler.handleCallback(Handler.java:978)
12-10 15:56:40.943 E/AndroidRuntime( 1846): 	at android.os.Handler.dispatchMessage(Handler.java:104)
12-10 15:56:40.943 E/AndroidRuntime( 1846): 	at android.os.Looper.loopOnce(Looper.java:238)
12-10 15:56:40.943 E/AndroidRuntime( 1846): 	at android.os.Looper.loop(Looper.java:357)
12-10 15:56:40.943 E/AndroidRuntime( 1846): 	at android.os.HandlerThread.run(HandlerThread.java:85)
```

**Comment by ellisonch** [comment]
Since the exception is coming from media.MediaMetadata, is it possible one of the strings you're supposed to be passing when you build one of these things is null?  Possibly here: https://github.com/VinylMusicPlayer/VinylMusicPlayer/blob/4e63e46796abaf35d30f065818fd80a32d4f8528/app/src/main/java/com/poupa/vinylmusicplayer/service/MusicService.java#L688C70-L688C70

As I said, I'm not an android developer, so this stuff is gibberish to me.  I just searched for things from the exception in your codebase.

I definitely see some stuff earlier in the log file related to missing strings:

```
12-10 15:56:40.790 W/Glide   (25038): There was 1 cause:
12-10 15:56:40.790 W/Glide   (25038): java.util.MissingResourceException(No artwork)
12-10 15:56:40.790 W/Glide   (25038):  call GlideException#logRootCauses(String) for more detail
12-10 15:56:40.790 W/Glide   (25038):   Cause (1 of 1): class com.bumptech.glide.load.engine.GlideException: Fetching data failed, class java.io.InputStream, LOCAL
12-10 15:56:40.790 W/Glide   (25038): There was 1 cause:
12-10 15:56:40.790 W/Glide   (25038): java.util.MissingResourceException(No artwork)
12-10 15:56:40.790 W/Glide   (25038):  call GlideException#logRootCauses(String) for more detail
12-10 15:56:40.790 W/Glide   (25038):     Cause (1 of 1): class java.util.MissingResourceException: No artwork
12-10 15:56:40.791 W/Glide   (25038): Load failed for com.poupa.vinylmusicplayer.glide.audiocover.SongCover@e9e982cc with size [349x349]
12-10 15:56:40.791 W/Glide   (25038): class com.bumptech.glide.load.engine.GlideException: Failed to load resource
12-10 15:56:40.791 W/Glide   (25038): There was 1 cause:
12-10 15:56:40.791 W/Glide   (25038): java.util.MissingResourceException(No artwork)
12-10 15:56:40.791 W/Glide   (25038):  call GlideException#logRootCauses(String) for more detail
12-10 15:56:40.791 W/Glide   (25038):   Cause (1 of 1): class com.bumptech.glide.load.engine.GlideException: Fetching data failed, class java.io.InputStream, LOCAL
12-10 15:56:40.791 W/Glide   (25038): There was 1 cause:
12-10 15:56:40.791 W/Glide   (25038): java.util.MissingResourceException(No artwork)
12-10 15:56:40.791 W/Glide   (25038):  call GlideException#logRootCauses(String) for more detail
12-10 15:56:40.791 W/Glide   (25038):     Cause (1 of 1): class java.util.MissingResourceException: No artwork
12-10 15:56:40.801 I/libPerfCtl(  814): xgfGetFPS pid:1846 fps:58
12-10 15:56:40.801 I/libPerfCtl(  814): xgfGetFPS pid:25038 fps:10
12-10 15:56:40.801 I/libPowerHal(  814): [perfNotifyAppState] pack:com.poupa.vinylmusicplayer.ci, act:com.poupa.vinylmusicplayer.ui.activities.MainActivity, state:5, pid:25038, uid:10548, fps:60
12-10 15:56:40.801 I/UxUtility(  814): notifyForegroundApp pack:com.poupa.vinylmusicplayer.ci, uid:10548
12-10 15:56:40.801 I/TouchUtility(  814): notifyAppState error = NULL
12-10 15:56:40.807 W/AString (  794): ctor got NULL, using empty string instead
12-10 15:56:40.807 I/Gyroscope(  794): write temp cali: [-17391, -1782, 735, -377781, 30937, 21277]
12-10 15:56:40.853 W/id3     (25038): Problem reading datatype within Frame Body:Unable to find null terminated string
12-10 15:56:40.854 W/id3     (25038): track.mp3:Invalid Frame:Unable to find null terminated string
12-10 15:56:40.855 W/id3     (25038): track.mp3:Found duplicate TDRC frame in invalid situation,discarding:TDRC
12-10 15:56:40.911 W/NotificationHistory( 1538): Attempted to add notif for locked/gone/disabled user 0
12-10 15:56:40.914 I/id3     (25038): track.mp3:Found padding starting at:1530
12-10 15:56:40.928 I/id3     (25038): track.mp3:Found padding starting at:1774
```

**Comment by ellisonch** [comment]
One more thing that I found that might help.  This bug looks very similar:

https://review.lineageos.org/c/LineageOS/android_frameworks_base/+/369567

The developer describes crashing systemui with the same exception.  It looks like they're addressing a bug in a recent android patch that causes this behavior.  My guess is they're addressing the bug that is causing our behavior.

Based on their description of the problem, perhaps you can avoid the crash by making sure any metadata object you pass to the OS is not null.

**Comment by soncaokim** [comment]
Hello, here is another custom build to workaround in the direction you suggested (with https://review.lineageos.org/c/LineageOS/android_frameworks_base/+/369567).

Can you please revert back once tested with this binary: https://github.com/VinylMusicPlayer/VinylMusicPlayer/suites/19013528402/artifacts/1112211884

**Comment by ellisonch** [comment]
Sadly, I'm getting the same behavior.

```
12-13 11:02:59.547 D/InterruptionStateProvider( 1856): No alerting: filtered notification: 0|com.poupa.vinylmusicplayer.ci|1|null|10549
12-13 11:02:59.551 D/InterruptionStateProvider( 1856): No alerting: filtered notification: 0|com.poupa.vinylmusicplayer.ci|1|null|10549
--------- beginning of crash
12-13 11:02:59.553 E/AndroidRuntime( 1856): FATAL EXCEPTION: SysUiBg
12-13 11:02:59.553 E/AndroidRuntime( 1856): Process: com.android.systemui, PID: 1856
12-13 11:02:59.553 E/AndroidRuntime( 1856): java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.media.MediaMetadata.getString(java.lang.String)' on a null object reference
12-13 11:02:59.553 E/AndroidRuntime( 1856): 	at com.android.systemui.media.MediaDataManager.loadMediaDataInBg(MediaDataManager.kt:580)
12-13 11:02:59.553 E/AndroidRuntime( 1856): 	at com.android.systemui.media.MediaDataManager.access$loadMediaDataInBg(MediaDataManager.kt:103)
12-13 11:02:59.553 E/AndroidRuntime( 1856): 	at com.android.systemui.media.MediaDataManager$loadMediaData$1.run(MediaDataManager.kt:343)
12-13 11:02:59.553 E/AndroidRuntime( 1856): 	at android.os.Handler.handleCallback(Handler.java:978)
12-13 11:02:59.553 E/AndroidRuntime( 1856): 	at android.os.Handler.dispatchMessage(Handler.java:104)
12-13 11:02:59.553 E/AndroidRuntime( 1856): 	at android.os.Looper.loopOnce(Looper.java:238)
12-13 11:02:59.553 E/AndroidRuntime( 1856): 	at android.os.Looper.loop(Looper.java:357)
12-13 11:02:59.553 E/AndroidRuntime( 1856): 	at android.os.HandlerThread.run(HandlerThread.java:85)
```

I looked at the code in your branch, and I see the change you made to `setMetadata()` at the start.  Could it be possible that MediaMetadataCompat.Builder().build() is returning null?  Its documentation (https://developer.android.com/reference/kotlin/android/support/v4/media/MediaMetadataCompat.Builder#build()) suggests it returns `MediaMetadataCompat!`, which if I understand right, means it could possibly be null.

Also, there are three calls to that function in MusicService.java.  In the error log I pasted in a previous comment, I see

```
12-13 11:02:59.417 W/Glide   (28938): Load failed for com.poupa.vinylmusicplayer.glide.audiocover.SongCover@3229e11f with size [349x349]
```
so I imagine it could get to the `onLoadFailed()` code.  Any of the three calls to `build()` could be returning null.

Is there a guide you can point to to explain how to turn the source into an APK?  I'd be happy to play around with it myself if I knew how to create the APK.

**Comment by ellisonch** [comment]
I'm not an android (or any kind of phone) developer.  I've never built any APK.  But, based on your suggestion, I installed Android Studio and loaded this project.  However, trying to run it gives me an error about the play store and toolchains.  The error seems to be in what Gradle is doing, not even actually building code.  After googling the errors, I can't seem to figure out how to fix it.

If you think it makes sense, I can post another issue related to not being able to build Vinyl.

**Comment by soncaokim** [comment]
> If you think it makes sense, I can post another issue related to not being able to build Vinyl.

Yes, please open a new issue.
If there is anything I can do to lower the barrier for a new joiner, I will do it
